x-pack/plugins/infra/public/alerting/metric_threshold/components/validation.tsx

```diff
@@ -63,6 +64,16 @@ export function validateMetricThreshold({
       );
     }
 
+    // The Threshold component returns an empty array with a length ([empty]) because it's using delete newThreshold[i].
+    // We need to use [...c.threshold] to convert it to an array with an undefined value ([undefined]) so we can test each element.
+    if (c.threshold && c.threshold.length && ![...c.threshold].every(isNumber)) {
+      errors[id].threshold0.push(
```

Comment:
I think we need to keep track of the actual threshold that's in error. It could be that `errors[id].threshold1` is in error.